File: tests/sources/basic/27-simple_python/test/simple.py
from pycompss.api.parameter import *
from pycompss.api.task import task

def main_program():
    from pycompss.api.api import compss_open

    # Check and get parameters
    if len(sys.argv) != 2:
        usage()
        exit(-1) 
    initialValue = sys.argv[1]
    fileName="counter"
    
    # Write value
    fos = open(fileName, 'w')
    fos.write(initialValue)
    fos.close()
    print "Initial counter value is " + initialValue
    
    # Execute increment
    increment(fileName)
    
    # Write new value
    fis = compss_open(fileName, 'r+')
    finalValue = fis.read()
    fis.close()
    print "Final counter value is " + finalValue
 
@task(filePath = FILE_INOUT)
def increment(filePath):
    print "Init task user code"
    # Read value
    fis = open(filePath, 'r')
    value = fis.read()
    print "Received " + value
    fis.close()

    # Write value
    fos = open(filePath, 'w')
    newValue=str(int(value) + 1)
    print "Computed " + newValue
    fos.write(newValue)
    fos.close()

def usage():
    print "[ERROR] Bad number of parameters"
    print "    Usage: simple <counterValue>"


if __name__ == "__main__":
    main_program()
    

File: tests/sources/java/3_remote_file/src/main/java/remoteFile/Simple.java
package remoteFile;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;


public class Simple {
    
    private static final String counterIN = "file://COMPSsWorker01@/tmp/COMPSsWorker01/counter";
    private static final String counterPathIN = "/tmp/COMPSsWorker01/counter";
    private static final String counterPathOUT = "counter";

    public static void main(String[] args) {
        // Check and get parameters
        if (args.length != 1) {
            System.out.println("[ERROR] Bad number of parameters");
            System.out.println("    Usage: simple.Simple <counterValue>");
            System.exit(-1);
        }
        
        int initialValue = Integer.parseInt(args[0]);

        // ------------------------------------------------------------------------
        // Write value
        try {
            FileOutputStream fos = new FileOutputStream(counterPathIN);
            fos.write(initialValue);
            System.out.println("Initial counter value is " + initialValue);
            fos.close();
        } catch (IOException ioe) {
            ioe.printStackTrace();
            System.exit(-1);
        }

        // ------------------------------------------------------------------------
        // Execute increment
        SimpleImpl.increment(counterIN, counterPathOUT);

        // ------------------------------------------------------------------------
        // Read new value
        System.out.println("After Sending task");
        try {
            FileInputStream fis = new FileInputStream(counterPathOUT);
            System.out.println("Final counter value is " + fis.read());
            fis.close();
        } catch (IOException ioe) {
            ioe.printStackTrace();
            System.exit(-1);
        }
    }

}
